design-proposals/feature-lifecycle.md

```diff
@@ -0,0 +1,327 @@
+  an undergraduate ("Beta") feature is planned to be removed.
+- As a KubeVirt cluster operator, I would like to stop using a feature
+  that got removed.
+
+### Repos
+This is a cross repo project policy under the
+[kubevirt](https://github.com/kubevirt) organization.
+
+## Proposal Design
+The proposal on how to define a feature lifecycle is influenced by
+processes and policies from the Kubernetes project.
+These sources are scattered around, each focusing on different
+aspects of a feature:
+- [Feature Gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
+- [Graduation](https://kubernetes.io/blog/2020/08/21/moving-forward-from-beta/)
+- [Changing the API](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md)
+- [Deprecation](https://kubernetes.io/docs/reference/using-api/deprecation-policy/)
+
+The proposal takes the top-down approach, starting with the high level
+flow that a common feature will traverse through.
+Continuing with actions that need to be taken and timeline suggestions.
+
+Both feature graduation and discontinuation flows are covered.
+Including the implications on users.
+
+Depending on individual topics, follow-up proposals may extend the basic
+points raised in this proposal.
+
+### Terminology
+Feature Gates and feature configuration are often used interchangeable or not differentiated.
+- Feature Gate: A flag that controls the presence or availability of a feature in the cluster.
+- Feature configuration: Cluster or workload level configuration that allows an admin or user
+  (depending on the feature) to control aspects of a feature operation.
+  A common usage is to determine if features are opt-in or opt-out by default.
+
+### Feature Stages
+A feature is expected to pass in the following order through the following stages:
+1. Enhancement proposal.
+2. Implementation.
+3. Release as Alpha (experimental).
+4. Release as Beta (pre-release for evaluation).
+5. Release as General Availability (graduation).
+6. Removal.
+
+Starting from the Alpha release, it can be removed with restrictions that
+depend on the release stage (Alpha, Beta, GA).
+
+[Removal](#removal) of features is widely discussed later
+in this proposal.
+
+#### Enhancement proposal
+As the first step for introducing a new feature, a formal proposal is
+expected to be shared for public review via mailinglist and
+a [design proposal](https://github.com/kubevirt/community/tree/main/design-proposals).
+
+This is the first opportunity to evaluate a new feature.
+The proposal needs to include motivation, goals, implementation details
+and phases. Review the [proposal template](https://github.com/kubevirt/community/blob/main/design-proposals/proposal-template.md)
+for more information.
+
+#### Implementation
+The development work on the feature is expected to include coding,
+testing, integration and documentation.
+
+#### Releases
+- **Alpha**:
```

Comment:
Should we strongly recommend that such features emit a warning?

Reply:
As is it requires a FG, I do not think it is not something needed.
Buy if we find in practice that it will help with some misuse, we can add it.

Reply:
Sounds reasonable.